backend/src/main/java/saci/application/KnowledgeController.java

```diff
@@ -68,4 +71,22 @@ public ResponseEntity<List<Knowledge>> getKnowledges() {
                 ? ResponseEntity.noContent().build()
                 : ResponseEntity.ok(knowledges);
     }
+
+    @Operation(
+            summary = "Delete knowledge by ID",
+            description = "Deletes a knowledge based on its ID")
+    @ApiResponses(
+            value = {
+                @ApiResponse(responseCode = "200", description = "Knowledge deleted successfully"),
+                @ApiResponse(responseCode = "404", description = "Knowledge not found")
+            })
+    @DeleteMapping("/{knowledgeId}")
+    public ResponseEntity<Void> deleteKnowledgeById(@PathVariable long knowledgeId) {
+        Optional<Knowledge> knowledge = knowledgeService.findById(knowledgeId);
```

Comment:
Please, move this logic to the service. The service should check if it exists or not. The controller should just call the action